boa/vyper/ast_utils.py

```diff
@@ -50,3 +50,18 @@ def ast_map_of(ast_node):
     for node in nodes:
         ast_map[getpos(node)] = node
     return ast_map
+
+
+def get_fn_name_from_lineno(ast_map, lineno: int):
+    def _walk_to_fn_node(_node):
+        if _node is None:
+            return ""
+        if _node.ast_type == "FunctionDef":
+            return _node.name
+        return _walk_to_fn_node(_node.get_ancestor())
+
+    for source_map, node in ast_map.items():
```

Comment:
i don't like the performance of this way of doing things

Reply:
It worked just about fine for contracts as big as tricrypto (2k+ lines of code).